src/SensorUtils.cpp

```
@@ -68,6 +72,116 @@ double PhaseAngle(const std::vector<double> &observerBodyFixedPosition,
 
 
 
+
+/**
+ * @brief rect2lat:  Converts rectanglur coordinates to [R,RightAscension,Declination]
+ *
+ * Rectanglular Cartesian Coordinates are related to Spherical Coordinates by:
+ *
+ * (1) X/R = cos(Declination)cos(RightAscension)
+ * (2) Y/R = cos(Declination)cos(RightAscension)
+ * (3) Z/R = sin(Declination)
+ *
+ * Where Z = [X^2 + Y^2 + Z^2]^(1/2)
+ *
+ * Dividing (2) by (1) and solving for RightAscension gives us:
+ *
+ * RightAscension = atan(Y/X)
+ *
+ * Declination can be solved directly:
+ *
+ * Declination = asin(Z/R)
+ *
+ *
+ * @author Tyler Wilson
+ * @param rectangularCoords
+ * @return  Given a set of J2000 coordinates, returns [R,RightAscension,Declination] in Radians
+ */
+vector<double> rect2lat(const vector<double> rectangularCoords){
+
+  vector<double> radiusLatLong{0.0,0.0,0.0};
+  vec coords(rectangularCoords);
+  double maxCoord = coords.max();
+
+  if (maxCoord > 0.0) {
```

It's much cleaner to have a check if maxCoord is approximately 0, then return a zero vector. This way, you don't have nested if statements and there's a clear special case being handled.

---

Do you mean the function should throw an exception if that is the case?

---

I think the current functionality of returning a zero vector is what we want to do, but it isn't clear from the code that inputting a zero vector will return a zero vector

---

This is what' I'm imagining

```
if (radius ~= 0) {
   return {0,0,0}
}

// Compute the lat and lon
```

That makes how we are handling the 0 input case clear and the lat, lon calculations don't have to be inside an if block.